tests/test_multi_standbys.py

```diff
@@ -243,13 +248,118 @@ def test_012_fail_primary():
+# primary is now blocking writes, we can set number-sync-standbys to 0 to
+# unblock writes, causing the primary to reach wait_primary
+#
+def test_014_001_fail_set_properties():
+    eq_(node1.get_number_sync_standbys(), 1)
+
+    node1.set_candidate_priority(50)
+    node2.set_candidate_priority(50)
+    node3.set_candidate_priority(50)
+
+    node1.wait_until_state(target_state="primary")
+
+    assert node1.get_replication_quorum()
+    assert node2.get_replication_quorum()
+    assert node3.get_replication_quorum()
+
+
+def test_014_002_fail_two_standby_nodes():
+    node2.fail()
+    node3.fail()
+
+    node2.wait_until_assigned_state(target_state="catchingup")
+    node3.wait_until_assigned_state(target_state="catchingup")
+
+    # node1 remains a primary, blocking writes, at this stage
+    node1.wait_until_state(target_state="primary")
+
+    ssn = "ANY 1 (pgautofailover_standby_2, pgautofailover_standby_3)"
+    eq_(node1.get_synchronous_standby_names(), ssn)
+    eq_(node1.get_synchronous_standby_names_local(), ssn)
+
+
+def test_014_003_unblock_writes():
+    node1.set_number_sync_standbys(0)
+
+    # node1 unblocks writes because number-sync-standbys is now zero
+    node1.wait_until_state(target_state="wait_primary")
+    eq_(node1.get_number_sync_standbys(), 0)
+
+    ssn = ""
+    eq_(node1.get_synchronous_standby_names(), ssn)
+    eq_(node1.get_synchronous_standby_names_local(), ssn)
+
+
+def test_014_004_restart_nodes():
+    node3.run()
+    node2.run()
+
+    assert node3.wait_until_state(target_state="secondary")
+    assert node2.wait_until_state(target_state="secondary")
+    assert node1.wait_until_state(target_state="primary")
+
+    ssn = "ANY 1 (pgautofailover_standby_2, pgautofailover_standby_3)"
+    eq_(node1.get_synchronous_standby_names(), ssn)
+    eq_(node1.get_synchronous_standby_names_local(), ssn)
+
+
+#
+# Now if number-sync-standbys is zero already, then when we lose all the
+# standby nodes the primary is assigned wait_primary to unblock writes
+#
+def test_015_001_set_properties():
+    node1.wait_until_state(target_state="primary")
+
+    eq_(node1.get_number_sync_standbys(), 0)
+
+
+def test_015_002_fail_two_standby_nodes():
+    node2.fail()
+    node3.fail()
+
+    # node 1 is assigned wait_primary as soon as we lose all the candidates
+    node1.wait_until_state(target_state="wait_primary")
+
+    ssn = ""
+    eq_(node1.get_synchronous_standby_names(), ssn)
+    eq_(node1.get_synchronous_standby_names_local(), ssn)
+
+
+def test_015_restart_nodes():
+    node3.run()
+    node2.run()
+
+    assert node3.wait_until_state(target_state="secondary")
+    assert node2.wait_until_state(target_state="secondary")
+    assert node1.wait_until_state(target_state="primary")
+
+    ssn = "ANY 1 (pgautofailover_standby_2, pgautofailover_standby_3)"
+    eq_(node1.get_synchronous_standby_names(), ssn)
+    eq_(node1.get_synchronous_standby_names_local(), ssn)
```

Comment:
I'm missing a test where we change increase `number-sync-standbys` to 1 to block the writes on the wait_primary.

Reply:
Fair. That said when doing that the wait_primary node is going to be apply_settings/primary and failing to reach primary for a while, so the `pg_autoctl set formation number-sync-standbys 1` command is going to time-out after 60s.

Reply:
Test added now, using `monitor.run_sql_query()` instead of the command line, to avoid the timeout and more failure modes to take into account.